src/v/cluster/types.h

```diff
@@ -1291,4 +1303,18 @@ struct adl<cluster::topic_properties> {
     void to(iobuf&, cluster::topic_properties&&);
     cluster::topic_properties from(iobuf_parser&);
 };
+
+template<typename T>
+struct adl<cluster::property_update<T>> {
+    void to(iobuf& out, cluster::property_update<T>&& update) {
+        reflection::serialize(out, std::move(update.value), update.op);
+    }
+
+    cluster::property_update<T> from(iobuf_parser& parser) {
+        auto value = reflection::adl<T>{}.from(parser);
```

Comment:
i defined it this way as the only difference between specialization for `tristate` and other types is default constructor. This will work for all the types

Reply:
I don't think there are any restrictions in place to prevent this from breaking. For example:

```c++
template<typename T>
struct property_update<Foo<T>> {
    NotAFoo<T> value;
    incremental_update_operation op = incremental_update_operation::none;
};
```

This won't work because `value` is not a `Foo<T>`. It should compile, but the on-wire format would be different if Foo and NotAFoo had different encoding protocol. Perhaps you need a static assertion that `update.value` is the same type as `T`?

Reply:
I will do it as a follow up PR